tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -296,6 +344,43 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						heapUsedStats.marginOfErrorPercent > args.maxRelativeMarginOfError
 					);
 
+					if (baselineMemoryUsage >= 0 && allowedDeviationBytes >= 0) {
+						// Compare the average heap used to the baseline memory usage
+						const avgHeapUsed = heapUsedStats.arithmeticMean;
+						const lowerBound = baselineMemoryUsage - args.allowedDeviationBytes;
+						const upperBound = baselineMemoryUsage + args.allowedDeviationBytes;
+						// Throw errors on regressions/improvements if `ENABLE_MEM_REGRESSION` is set and a warning otherwise.
+						// This allows us to run the tests in CI without failing the build, but still get a warning.
+						if (avgHeapUsed > upperBound) {
+							const message = `Memory Regression detected for test '${
+								testObject.title
+							}': Used '${avgHeapUsed.toPrecision(6)}' bytes, with baseline'${
+								args.baselineMemoryUsage
+							}' and tolerance of '${allowedDeviationBytes} bytes.\n`;
+							if (ENABLE_MEM_REGRESSION) {
+								process.stderr.write(message);
+								// eslint-disable-next-line unicorn/no-process-exit
+								process.exit(1);
```

Comment:
Thanks for the screenshots! They did showcase a problem we need to fix: when `ENABLE_MEM_REGRESSION=1` we still need to see the table output like when it is set to 0, including any other tests that might have run, and the regressed test should be marked as failed in there. Right now this makes it so as soon as a test is determined to have regressed, we just terminate the whole process, run no further tests, and get no output for anything else.

I think just `throw new Error(message)` here should fix it... all this is already wrapped in a try-catch that sets `benchmarkStats = { error: (error as Error).message };` and should result in the expected behavior (the specific test ending up as failed in the output).

Reply:
@alexvy86 I had tried this. `throw new Error(message)` just causes the test to fail silently since it's the child process throwing the error because of this line. 

```typescript
if (result.stderr !== "") {
	assert.fail(`Child process logged errors: ${result.stderr}`);
}
```

From what I can tell, Mocha catches the error but doesn't surface it back to the parent process.

Reply:
There's something we need to look into in that case though, because I'm almost certain we follow a similar pattern for the other kinds of benchmarks and that's how they behave. The child process should be printing errored tests to `stdout` somehow, for the parent process to then aggregate results and print the output.